### PR TITLE
Copy nvjpeg utils into imgcodec

### PR DESCRIPTION
**Pull request by staniewzki**
Signed-off-by: Michał Staniewski <m.staniewzki@gmail.com>

## Category:
**Refactoring** (*Redesign of existing code that doesn't affect functionality*)

## Description:

Preparing files needed for nvjpeg2k codec (https://github.com/NVIDIA/DALI/pull/4114), copying them from `dali/operators/decoder/nvjpeg`.  Changed names of classes in `nvjpeg2k_helper.h` to make them consistent.

### Affected modules and functionalities:

N/A

### Key points relevant for the review:

N/A

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by jantonguirao — APPROVED** [review]

File: dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.cc [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  case NVJPEG2K_STATUS_SUCCESS:
+    return;
+  default:
+    throw dali::imgcodec::NvJpeg2kError(status, extra.c_str());
+  }
+}
+
+}  // namespace dali
+
+#endif  // DALI_IMGCODEC_DECODERS_NVJPEG_NVJPEG2K_HELPER_H_
```
Comment (klecki) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.cc:156]: This won't pass linter, please fix the newlines and the header guard (it doesn't match the name of the file).

**Review by jantonguirao — APPROVED** [review]

File: dali/imgcodec/decoders/nvjpeg/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+collect_headers(DALI_INST_HDRS PARENT_SCOPE)
+collect_sources(DALI_IMGCODEC_SRCS PARENT_SCOPE)
+collect_test_sources(DALI_IMGCODEC_TEST_SRCS PARENT_SCOPE)
```
Comment (klecki) [inline dali/imgcodec/decoders/nvjpeg/CMakeLists.txt:17]: New line please.

File: dali/imgcodec/decoders/nvjpeg/nvjpeg2k_helper.h [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  static void DestroyHandle(nvjpeg2kHandle_t handle) {
+    nvjpeg2kDestroy(handle);
+  }
+};
+
+struct NvJpeg2kStream : public UniqueHandle<nvjpeg2kStream_t, NvJpeg2kStream> {
+  DALI_INHERIT_UNIQUE_HANDLE(nvjpeg2kStream_t, NvJpeg2kStream);
+
+  static NvJpeg2kStream Create() {
```
Comment (klecki) [inline dali/imgcodec/decoders/nvjpeg/nvjpeg2k_helper.h:105]: Why this one uses Crate and the others regular constructors?

**Review by klecki — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Comment by staniewzki** [comment]
!build

**Pull request merged**